src/csharp/Grpc.HealthCheck/HealthGrpc.cs

```diff
@@ -38,11 +38,22 @@ public interface IHealthClient
     }
 
     // server-side interface
+    [System.Obsolete("Service implementations should inherit from the generated abstract base class instead.")]
     public interface IHealth
     {
       Task<global::Grpc.Health.V1.HealthCheckResponse> Check(global::Grpc.Health.V1.HealthCheckRequest request, ServerCallContext context);
     }
 
+    // server-side abstract class
+    public abstract class HealthBase
```

Comment:
Doing them in the same PR (but different commits) makes sense to me - and the benefit of not having an obsolete interface is big, IMO.